pkg/analyzer/analyzers/postgres/postgres.go

```diff
@@ -14,8 +16,168 @@ import (
+		return nil, err
+	}
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+func secretInfoToAnalyzerResult(info *SecretInfo) *analyzers.AnalyzerResult {
+	if info == nil {
+		return nil
+	}
+	result := analyzers.AnalyzerResult{
+		AnalyzerType: analyzerpb.AnalyzerType_Postgres,
+		Metadata:     nil,
+		Bindings:     []analyzers.Binding{},
+	}
+
+	// set user related bindings in result
+	userResource, userBindings := bakeUserBindings(info)
+	result.Bindings = append(result.Bindings, userBindings...)
+
+	// add user's database priviliges to bindings
+	dbNameToResourceMap, dbBindings := bakeDatabaseBindings(userResource, info)
+	result.Bindings = append(result.Bindings, dbBindings...)
+
+	// add user's table priviliges to bindings
+	tableBindings := bakeTableBindings(dbNameToResourceMap, info)
+	result.Bindings = append(result.Bindings, tableBindings...)
+
+	return &result
+}
+
+func bakeUserBindings(info *SecretInfo) (analyzers.Resource, []analyzers.Binding) {
+	userResource := analyzers.Resource{
+		Name:               info.User,
+		FullyQualifiedName: info.User,
+		Type:               "user",
+		Metadata: map[string]any{
+			"role": info.Role,
+		},
+	}
+
+	var bindings []analyzers.Binding
+
+	for rolePriv, exists := range info.RolePrivs {
+		if exists {
+			bindings = append(bindings, analyzers.Binding{
+				Resource: userResource,
+				Permission: analyzers.Permission{
+					Value: rolePriv,
+				},
+			})
+		}
+	}
+
+	return userResource, bindings
+}
+
+func bakeDatabaseBindings(userResource analyzers.Resource, info *SecretInfo) (map[string]*analyzers.Resource, []analyzers.Binding) {
+	dbNameToResourceMap := map[string]*analyzers.Resource{}
+	dbBindings := []analyzers.Binding{}
+
+	for _, db := range info.DBs {
+		dbResource := analyzers.Resource{
+			Name:               db.DatabaseName,
+			FullyQualifiedName: db.DatabaseName,
+			Type:               "database",
+			Metadata: map[string]any{
+				"owner": db.Owner,
+			},
+			Parent: &userResource,
+		}
+
+		// populate map to reference later for tables
+		dbNameToResourceMap[db.DatabaseName] = &dbResource
+
+		dbPriviliges := map[string]bool{
+			"connect": db.Connect,
+			"create":  db.Create,
+			"temp":    db.CreateTemp,
+		}
+
+		for priv, exists := range dbPriviliges {
+			if exists {
+				dbBindings = append(dbBindings, analyzers.Binding{
+					Resource: dbResource,
+					Permission: analyzers.Permission{
+						Value: priv,
+					},
+				})
+			}
+		}
+	}
+
+	return dbNameToResourceMap, dbBindings
+}
+
+func bakeTableBindings(dbNameToResourceMap map[string]*analyzers.Resource, info *SecretInfo) []analyzers.Binding {
+	var tableBindings []analyzers.Binding
+
+	for dbName, tableMap := range info.TablePrivs {
+		dbResource, ok := dbNameToResourceMap[dbName]
+		if !ok {
+			continue
+		}
+
+		for tableName, tableData := range tableMap {
+			tableResource := analyzers.Resource{
+				Name:               tableName,
+				FullyQualifiedName: dbResource.Name + "." + tableName,
```

Comment:
Is there a reason you use `.` here instead of `/`? I thought `/` was the standard for separating fields in the fully qualified name.

Also, after reviewing the MySQL analyzer I _think_ we may need to make this more unique to include the URI or hostname?

Reply:
@mcastorina Usually, in queries, tables are reference with database name appended with '.' To have similarity with that, I made fully qualified name. We can go for '/' as well.

Hostname seems to be a good idea to make fully qualified name unique.

Reply:
Ah, if it makes more sense to use `.` then go for it, I just wasn't aware of that idiom!

Reply:
I'm going to go ahead and merge now, but feel free to open another PR if you'd like to make that change.